--- a/src/gui/calendar-management/gcal-new-calendar-page.c
+++ b/src/gui/calendar-management/gcal-new-calendar-page.c
@@ -352,7 +352,7 @@
 
   self->validate_url_resource_id = 0;
 
-  guri = g_uri_parse (gtk_entry_get_text (GTK_EDITABLE (self->calendar_address_entry)), SOUP_HTTP_URI_FLAGS | G_URI_FLAGS_PARSE_RELAXED, NULL);
+  guri = g_uri_parse (gtk_entry_get_text (GTK_ENTRY (self->calendar_address_entry)), SOUP_HTTP_URI_FLAGS | G_URI_FLAGS_PARSE_RELAXED, NULL);
 
   if (guri != NULL && g_uri_get_scheme (guri) && g_uri_get_host (guri))
     {
